### PR TITLE
Create demographic-research.csl

### PR DESCRIPTION
**Pull request by zuphilip**
@ikashnitsky started and I completed the work for the OA journal Demographic Research.

See https://forums.zotero.org/discussion/57130/style-request-demographic-research/

### CODE REVIEW

File: demographic-research.csl [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+          <number variable="edition" form="ordinal"/>
+          <text term="edition" form="short"/>
+        </group>
+      </if>
+      <else>
+        <text variable="edition" suffix="."/>
+      </else>
+    </choose>
+  </macro>
+  <citation et-al-min="4" et-al-use-first="1" disambiguate-add-names="true" disambiguate-add-givenname="true" disambiguate-add-year-suffix="true" collapse="year">
```
Comment (adam3smith) [inline demographic-research.csl]: let's sort these by year-date, as suggested by the examples for multiple works by same author.

Comment (adam3smith) [inline demographic-research.csl]: actually, sorry, do alphabetical by author, then by year. See: http://www.demographic-research.org/volumes/vol34/12/34-12.pdf

File: demographic-research.csl [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+          <text macro="author-short"/>
+          <text macro="year-date"/>
+        </group>
+        <text variable="locator"/>
+      </group>
+    </layout>
+  </citation>
+  <bibliography hanging-indent="true" entry-spacing="0" line-spacing="1">
+    <sort>
+      <key macro="author"/>
```
Comment (adam3smith) [inline demographic-research.csl]: let's sort by year-date before title. It's not clear either way in the guide, but year is more typical and confirmed by the two Schnell citations in http://www.demographic-research.org/volumes/vol34/11/34-11.pdf

File: demographic-research.csl [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            <text variable="volume" font-weight="normal" prefix=" "/>
+            <text variable="issue" prefix="(" suffix=")"/>
+            <text variable="page" prefix=":"/>
+          </group>
+        </else>
+      </choose>
+      <text macro="access" prefix=". "/>
+    </layout>
+  </bibliography>
+  <locale xml:lang="en-GB">
```
Comment (adam3smith) [inline demographic-research.csl]: let's move this up under the end of the info section for consistency, please

**Comment by ikashnitsky** [comment]
Looks nice!

(I'm not sure whether it's a job for CSL)
DOIs are represented as hyperlinks in DemRes papers
Is it possible / necessary to change this?

**Comment by adam3smith** [comment]
CSL can't insert hyperlinks. I'd very much expect DemRes' publishing system to handle that automatically.

**Comment by ikashnitsky** [comment]
Got it.

Thanks a lot for your input

**Comment by adam3smith** [comment]
great, thanks!

**Pull request merged**